pkg/operator/starter.go

```diff
@@ -50,62 +52,95 @@ const (
 	infrastructureName = "cluster"
 )
 
-func RunOperator(ctx context.Context, controllerConfig *controllercmd.ControllerContext) error {
-	// Create core clientset and informer
-	kubeClient := kubeclient.NewForConfigOrDie(rest.AddUserAgent(controllerConfig.KubeConfig, operatorName))
-	kubeInformersForNamespaces := v1helpers.NewKubeInformersForNamespaces(kubeClient, defaultNamespace, cloudConfigNamespace, "")
-	secretInformer := kubeInformersForNamespaces.InformersFor(defaultNamespace).Core().V1().Secrets()
-	nodeInformer := kubeInformersForNamespaces.InformersFor("").Core().V1().Nodes()
-	configMapInformer := kubeInformersForNamespaces.InformersFor(defaultNamespace).Core().V1().ConfigMaps()
-
-	// Create config clientset and informer. This is used to get the cluster ID
-	configClient := configclient.NewForConfigOrDie(rest.AddUserAgent(controllerConfig.KubeConfig, operatorName))
-	configInformers := configinformers.NewSharedInformerFactory(configClient, 20*time.Minute)
-	infraInformer := configInformers.Config().V1().Infrastructures()
-
-	// Create informer for the ConfigMaps in the openshift-config-managed namespace. This is used to get the custom CA
-	// bundle to use when accessing the AWS API.
-	cloudConfigInformer := kubeInformersForNamespaces.InformersFor(cloudConfigNamespace).Core().V1().ConfigMaps()
-	cloudConfigLister := cloudConfigInformer.Lister().ConfigMaps(cloudConfigNamespace)
-
-	// Create GenericOperatorclient. This is used by the library-go controllers created down below
-	gvr := opv1.SchemeGroupVersion.WithResource("clustercsidrivers")
-	operatorClient, dynamicInformers, err := goc.NewClusterScopedOperatorClientWithConfigName(controllerConfig.KubeConfig, gvr, instanceName)
+func RunOperator(ctx context.Context, controllerConfig *controllercmd.ControllerContext, guestKubeConfigString *string) error {
```

Comment:
Why is `guestKubeConfigString` a pointer?

Reply:
Fixed